source/ui/netsender.c

```diff
@@ -0,0 +1,607 @@
+#include "netsender.h"
+
+#include <malloc.h>
+#include <sys/socket.h>
+#include <arpa/inet.h>
+#include <netinet/in.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <netdb.h>
+
+#include <zlib.h>
+
+#define ZLIB_CHUNK (16 * 1024)
+
+typedef u32 in_addr_t;
+
+static int datafd = -1;
+static struct in_addr dsaddr;
+static LightEvent event;
+static volatile size_t filelen, filetotal;
+static volatile bool wantExit = false;
+static volatile bool doneSearching = false;
+#define RECEIVER_IP_LENGTH 15
+static char receiverIp[RECEIVER_IP_LENGTH+1];
+
+static void netsenderError(const char* func, int err);
+
+static bool netsenderInit(void)
```

Comment:
I wonder if netsenderInit/netsenderDeactivate could be conflated with their netloader* counterparts since they have the same code.